dev/tools/dartdoc.dart

```diff
@@ -300,30 +302,46 @@ void cleanOutSnippets() {
   }
 }
 
-File precompileSnippetsTool() {
+Future<File> precompileSnippetsTool(
+  String pubExecutable,
+  Map<String, String> pubEnvironment,
+) async {
   final File snapshotPath = File(path.join('bin', 'cache', 'snippets.snapshot'));
   print('Precompiling snippets tool into ${snapshotPath.absolute.path}');
   if (snapshotPath.existsSync()) {
     snapshotPath.deleteSync();
   }
+
+  Process process = await Process.start(
+    pubExecutable,
+    <String>['get'],
+    workingDirectory: kSnippetsRoot,
+    environment: pubEnvironment,
+  );
+  printStream(process.stdout, prefix: 'pub:stdout: ');
+  printStream(process.stderr, prefix: 'pub:stderr: ');
+  final int code = await process.exitCode;
```

Comment:
Don't you also want to `await process.stdout` and `await process.stderr` so that you're sure it completes before exiting?

See what I had to do [here](https://github.com/flutter/flutter/blob/5dac7a23479c84dbe6b132815523b66a75dfd148/dev/bots/prepare_package.dart#L125) and then [added this](https://github.com/flutter/flutter/blob/5dac7a23479c84dbe6b132815523b66a75dfd148/dev/bots/prepare_package.dart#L166).